jenkins-plugin/src/main/resources/org/jenkinsci/plugins/pipeline/maven/MavenConfigFolderOverrideProperty/config.jelly

```diff
@@ -0,0 +1,31 @@
+<!--
+The MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+<f:section title="${%Pipeline Maven Configuration}">
+      <f:optionalBlock title="${%Override global maven configuration}" field="override" inline="true">

```

Comment:
OK `override` is used here. I think this could be done using something like `checked=${instance.isOverridden()}` to be able get rid of that `override` attribute.

Reply:
I want that attribute, makes things easier. If I remove it it means that I would have to provide a way to unset the combos...

Reply:
@alvarolobato did you offer the flexibility to override or not the Global Tools Configuration? Isn't it more sophisticated than the other folder level settings in Jenkins? My understanding is that folder level settings always override Jenkins Global Settings.

> "Override global maven configuration"

Is there a risk of misunderstanding on the meaning of "global"? "Maven global settings / settings" and "Global Tools Configuration level config" / "Folder level config"?

Shouldn't we say "Override maven configuration defined in the 'Global Tools Configuration'"? If this is this type of override, what about maven configuration defined in a parent folder?

Reply:
@cyrille-leclerc you can override both or only one of them, whatever the user decides.

> Isn't it more sophisticated than the other folder level settings in Jenkins? My understanding is that folder level settings always override Jenkins Global Settings.

Not sure what you mean with this. The user can choose to override and select one value from the combo for "Maven Settings" and "Maven Global Settings". This is the exact same behavior we have for Jenkins configuration and for Job Level configuration. When you override this in a Job, you are presented with the same as now in the folder, so I don't think this is more sophisticated.

>Shouldn't we say "Override maven configuration defined in the 'Global Tools Configuration'"? If this is this type of override, what about maven configuration defined in a parent folder?

I can change it, although I think it may be overkill and just removing "global" should be enough. The job settings don't even say anything:
<img width="1053" alt="screen shot 2018-01-15 at 13 11 06" src="https://user-images.githubusercontent.com/17157380/34941828-a9e85c4c-f9f5-11e7-98d1-343546adb70b.png">